test_integration/lib/test/realtime/realtime_auth_url_test.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:ably_flutter/ably_flutter.dart';
+import 'package:ably_flutter_integration_test/app_provisioning.dart';
+import 'package:ably_flutter_integration_test/factory/reporter.dart';
+
+//Make sure that creating realtime with authUrk connects without a problem
+Future<Map<String, dynamic>> testCreateRealtimeWithAuthUrl({
+  required Reporter reporter,
+  Map<String, dynamic>? payload,
+}) async {
+  reporter.reportLog('init start');
+  final appKey = await AppProvisioning().provisionApp();
+
+  final clientOptionsForToken = ClientOptions(
+    key: appKey,
+    environment: 'sandbox',
+    logLevel: LogLevel.verbose,
+  );
+
+  final ablyForToken = Rest(
+    options: clientOptionsForToken,
+  );
+  final tokenDetails = await ablyForToken.auth.requestToken();
+
+  final authUrl =
+      'https://echo.ably.io/?body=${Uri.encodeComponent(tokenDetails.token!)}';
+  final options = ClientOptions(
+      authUrl: authUrl,
+      environment: 'sandbox',
+      useTokenAuth: true,
+      autoConnect: false,
+      logLevel: LogLevel.verbose);
+  final realtime = Realtime(options: options);
+  var connected = false;
+  realtime.connection.on().listen((stateChange) {
+    if (stateChange.event == ConnectionEvent.connected) {
+      connected = true;
+    }
+  });
+  await realtime.connect();
+  await Future<void>.delayed(const Duration(seconds: 5));
+  if (!connected) {
+    throw Error();
+  }
```

Comment:
It's a bit hacky just waiting for 5 seconds and then checking if we're connected - would it be possible to use a completion listener to wait for the connected event instead:
```dart
final completer = Completer<void>();
realtime.connection.on().listen((stateChange) {
  if (stateChange.current == ConnectionState.connected) {
    completer.complete();
  }
});
await realtime.connect();
await completer.future;
```

Reply:
That's a good idea. I made change 3de55d70531ed8fb7c8e92f78831181f574a9a6b . This is with 30 seconds timeout, otherwise test wouldn't finish

Reply:
nice, thanks!